File: javascript/leetcode/remove-element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var i = 0
    var l = nums.length
    while (i < l) {
        if (nums[i] == val) {
            nums.splice(i, 1)
            l--
        } else {
            i++            
        }
    }
    return l
};

File: ruby/leetcode/remove-element.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
    i = 0
    l = nums.size()
    while i < l
        if nums[i] == val
            nums.delete_at(i)
            l -= 1
        else
            i += 1
        end
    end
    return l
end
